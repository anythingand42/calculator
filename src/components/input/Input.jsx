import styled from 'styled-components';

const StyledInput = styled.input`
  border: 1px solid #ccc;
  outline: none;
  box-sizing: border-box;
  padding: 4px;
  min-height: 28px;
  ${(props) => (props.error ? 'border-color: red; color: red;' : '')}
`;

export default StyledInput;
