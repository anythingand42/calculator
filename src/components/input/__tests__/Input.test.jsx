import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Input from '../Input';

test('Input component', () => {
  const { container } = render(<Input />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Input component with error prop', () => {
  const { container } = render(<Input error />);
  expect(container.firstChild).toMatchSnapshot();
});
