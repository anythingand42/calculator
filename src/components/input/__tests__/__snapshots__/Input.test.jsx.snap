// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 1`] = `
.c0 {
  border: 1px solid #ccc;
  outline: none;
  box-sizing: border-box;
  padding: 4px;
  min-height: 28px;
}

<input
  class="c0"
/>
`;

exports[`Input component with error prop 1`] = `
.c0 {
  border: 1px solid #ccc;
  outline: none;
  box-sizing: border-box;
  padding: 4px;
  min-height: 28px;
  border-color: red;
  color: red;
}

<input
  class="c0"
/>
`;
